module/dvfs/src/mod_dvfs.c

```diff
@@ -1127,13 +1127,17 @@ static int dvfs_bind_element(fwk_id_t domain_id, unsigned int round)
 
     /* Bind to the alarm HAL if required */
     if (ctx->config->retry_ms > 0) {
```

Comment:
If the timer is not used you can set config->retry_ms = 0 this will skip using the timer.

Reply:
when compiling without timer, FWK_MODULE_IDX_TIMER is undefined and as a result MOD_TIMER_API_ID_ALARM is undefined too

Reply:
You have 2 options:
1. Include the timer module but not use it.
2. #if BUILD_HAS_MOD_TIMER all of the timer related structures, variables and functions in dvfs module. This includes the config structure in dvfs header file.

I am not sure how this patch compiles if you are not including the timer.

Reply:
I don't know what build environment you are using but this will not work for our build system. Even your last patch is not enough.
I think the easiest way for you just to include the timer module without using it. We have examples in the code base like this[ product config file](https://github.com/ARM-software/SCP-firmware/blob/master/product/rdn2/scp_romfw/config_clock.c) The clock is included to gain access to the header file. This is option 1.
OR if you want NOT to include the timer then you need to #if all the timer related strcut,func,var in the dvfs .c and .h files